util/encoding.go

```diff
@@ -0,0 +1,31 @@
+package util
+
+import (
+	"bytes"
+	"encoding/binary"
+)
+
+// Helpers functions regarding the encoding of basic types ( integers ...)
```

Comment:
When encoding/decoding, especially in size-sensitive binary formats, I think it's best to use only the size-specific types (e.g., int64 or uint64) to make it abundantly clear from the code exactly what the size and hence encoding is.  For Varint and other variable-length encodings it might make more sense to use encode/decode functions that just take 'int' types.